.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,81 @@
+name: Run tests
+on:
+    push:
+
+jobs:
+  build-linux-cpp:
+    name: Run tests on ubuntu-18.04
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        compiler:
+          - CC: clang
+            CXX: clang++
+          - CC: gcc
+            CXX: g++
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          submodules: 'true'
+
+      - name: Install packages
+        run: sudo apt-get update && sudo apt-get install libhdf5-dev
+
+      - name: Build and run unittests
+        env:
+          CC: ${{ matrix.compiler.CC }}
+          CXX: ${{ matrix.compiler.CXX }}
+        run: |
+          ./ci/cpp_test.sh
+
+  build-linux-python:
+    name: Run python tests on ubuntu-18.04
+    runs-on: ubuntu-18.04
+    needs: build-linux-cpp # don't bother with python until cpp passes
+
+    strategy:
+      matrix:
+        python-version: ['3.6', '3.7', '3.8', '3.9', ]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          submodules: 'true'
+
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install packages
+        run: sudo apt-get update && sudo apt-get install libhdf5-dev doxygen
+
+      - name: Build and run unittests
+        run: |
+            ./ci/python_test.sh
+
+  build-macos:
+    name: Run tests on macos
+    runs-on: macos-latest
+    needs: build-linux-python # don't bother with mac until linux passes
```

Comment:
Imo, the shorter the CI duration the better for the devs. I don't see the point of this.

Reply:
I assume they'll throttle number of runners at some point.
Do you know if they send an email as soon as one job fails, or only after all have finished?

Reply:
If A and B are running in parallel, if A fails, it will abort B so I guess the mail is sent asap.

Source:
 The Linux build https://github.com/BlueBrain/MorphIO/runs/1656377247?check_suite_focus=true#step:5:2575
which was aborted because the Mac OS build failed (because of the exact same issue you solved in this PR)

Reply:
cool, good to know, thanks.